docs/base/explanation/parts.rst

```diff
@@ -0,0 +1,186 @@
+.. _parts:
+
+Parts
+=====
+
+In the Craft Parts framework, parts are descriptions of the components to be
+built and prepared for deployment in a payload, either individually or as
+part of a larger project containing many components.
+
+When the Craft Parts framework is used to process a part on behalf of a tool
+or library, it performs some or all of the steps described in the
+:ref:`parts lifecycle <lifecycle>`:
+
+#. The *pull* step pulls the source code and dependencies from locations
+   defined in the part and places them into a package cache.
+#. The *overlay* step unpacks them into a base file system chosen from a
+   collection of standard file system images.
+#. The *build* step runs a suitable build tool for the sources to compile
+   a set of build products or artefacts.
+#. The *stage* step copies the build products for the part into a common
+   area for all parts in a project.
+#. The *prime* step copies the files to be deployed into an area for
+   further processing.
+
+Not all of these steps may be needed for every use case, and tools that use
+the Craft Parts framework can skip those that are not appropriate for their
+purposes.
+
+Tools like `Snapcraft`_ and `Charmcraft`_ that use the concepts of parts to
+describe a build process typically accept specifications of parts in YAML format. This allows each part to be described in a convenient,
+mostly-declarative format. Libraries that use parts may use the underlying
+data structures to describe them.
+
+Describing a part
```

Comment:
Thinking about charmcraft docs, these sections would be nice as their own snippets if we were using sphinx.

We're not using sphinx for charmcraft so I'm not requesting anything right now, but it's a possible future improvement to separate them into snippets that can each be bookended with app-specific stuff.

Reply:
Never mind - I see how you're doing it in Rockcraft and that makes a lot of sense :-)